contracts/sentinel/guards/FuseWithdrawalGuard.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "../IGuard.sol";
+import "../../refs/CoreRef.sol";
+import "../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+import "../../pcv/PCVGuardian.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+contract FuseWithdrawalGuard is IGuard, CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    struct WithdrawInfo {
+        address destination;
+        address underlying;
+        uint96 liquidityToLeave;
+    }
+
+    /// @notice map the destination and minimum liquidity for each pcv deposit
+    mapping(address => WithdrawInfo) public withdrawInfos;
+
+    EnumerableSet.AddressSet private fuseDeposits;
+
+    /// @notice the PCV mover contract exposed to guardian role
+    PCVGuardian public constant pcvGuardian = PCVGuardian(0x02435948F84d7465FB71dE45ABa6098Fc6eC2993);
+
+    constructor(
+        address core,
+        address[] memory deposits,
+        address[] memory destinations,
+        address[] memory underlyings,
+        uint96[] memory liquidityToLeaveList
+    ) CoreRef(core) {
+        uint256 len = deposits.length;
+        require(len == destinations.length && len == liquidityToLeaveList.length && len == underlyings.length);
+        for (uint256 i = 0; i < len; ) {
+            fuseDeposits.add(deposits[i]);
+            withdrawInfos[deposits[i]] = WithdrawInfo({
+                destination: destinations[i],
```

Comment:
Is it worthwhile checking that all destinations are `safeAddresses`? `pcvGuardian.isSafeAddress()`? Then can guarantee we can recover the transferred out funds and not make a mistake by setting an incorrect address

Reply:
PCV Guardian can only transfer to a safe address so this is baked in